src/lib.rs

```diff
@@ -2944,6 +3207,18 @@ pub struct FormattingOptions {
     /// Signature for further properties.
     #[serde(flatten)]
     pub properties: HashMap<String, FormattingProperty>,
+
+    /// Trim trailing whitespaces on a line.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub trailing_whitespace: Option<bool>,
+
+    /// Insert a newline character at the end of the file if one does not exist.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub insert_final_new_line: Option<bool>,
```

Comment:
I think this needs to be `insert_final_newline` to match the casing in the spec: `insertFinalNewline`